napari_animation/_tests/test_animation.py

```diff
@@ -139,3 +145,45 @@ def test_end_state_reached(image_animation):
     image_animation.capture_keyframe(steps=2)
     last_state = image_animation._frames[-1]
     assert last_state == image_animation.key_frames[-1].viewer_state
+
+
+@pytest.mark.parametrize("layer_class, data, ndim", layer_test_data)
+def test_attributes_for_all_layer_types(
+    make_napari_viewer, layer_class, data, ndim
+):
+    """Test that attributes are in the viewer_state for all napari layer types"""
+    viewer = make_napari_viewer()
+    add_layer_by_type(viewer, layer_class, data, visible=True)
+    layer_animation = Animation(viewer)
+    # get the state of the layer
+    layer_state = viewer.layers[0]._get_state()
+    # remove attributes that arn't captured
+    layer_state.pop("metadata")
+    layer_state.pop("data", None)
+
+    if layer_class == Surface:
+        layer_state["normals"]["face"].pop("mode", None)
+        layer_state["normals"]["vertex"].pop("mode", None)
```

Comment:
I'm confused about this — is this something users would need to do if trying to animate a Surface layer? That seems clunky. Basically, the things in the test should mirror what users would do, and if it's some weird workaround, then it might be good to e.g. push those tests to a different xfail test, so that it's obvious what needs to be fixed in the codebase going forward...

Reply:
So the issue is, as Lorenzo noted, that these are enums flagged as immutable, so they arn't settable. So the layer state has them, but the `viewer_state` does not. The test compares the two sets of dicts, so I pop them from the layer state dict before the comparison check.
I don't think a user will ever run into this because they are immutable they can't changed during an animation anyways.

Reply:
Ha! Turns out I had a horrible logic bug, see:
https://forum.image.sc/t/issue-saving-animations-in-napari-animation/88868/8?u=psobolewskiphd
And with it fixed this hack isn't needed. I was setting only the states that *didn't* change, which would be the immutable ones.
Now when we set the changed ones, everything should be peachy!